pkg/controller/statefulset/stateful_set_control.go

```diff
@@ -224,17 +224,19 @@ func (ssc *defaultStatefulSetControl) getStatefulSetRevisions(
 	equalRevisions := history.FindEqualRevisions(revisions, updateRevision)
 	equalCount := len(equalRevisions)
 
-	if equalCount > 0 && history.EqualRevision(revisions[revisionCount-1], equalRevisions[equalCount-1]) {
+	if equalCount > 0 {
 		// if the equivalent revision is immediately prior the update revision has not changed
-		updateRevision = revisions[revisionCount-1]
-	} else if equalCount > 0 {
-		// if the equivalent revision is not immediately prior we will roll back by incrementing the
-		// Revision of the equivalent revision
-		updateRevision, err = ssc.controllerHistory.UpdateControllerRevision(
-			equalRevisions[equalCount-1],
-			updateRevision.Revision)
-		if err != nil {
-			return nil, nil, collisionCount, err
+		if history.EqualRevision(revisions[revisionCount-1], equalRevisions[equalCount-1]) {
```

Comment:
The code avoided nesting. How is this PR helping our code base other then presenting different opinion on the structure then the original author?

Reply:
thanks @tnozicka 
Good question !
This PR is to change the struct like

```
if A && B {
	
} else if A {
	
} else {
	
}
```

to the struct like

```
if A {
	if B {

	} else {

	}
} else {
	
}
```

From Boolean logic, we can be sure that there will be no moral change. 
1. Such nesting, we can easy to see that the outer `if... else...` is to judge whether `A ` is true, and the inner is to judge whether `B` is true on this basis. In the original way, it is not easy to judge the exact logical meaning represented by else if `A`.
2. When `A` is false, the original writing method needs to judge if and else if, because them both contain `A` , while PR's writing method does not, it only needs to be judged once.

So I think this PR is more neat and understandable.

Reply:
I agree with Tomas here, although from logical point of view nothing changes, the readability actually hurts. The preferred approach is to have less nesting than needed. Your code goes the other direction. I could be convinced in favor of this change, but only if the calculation of `A` from your example was computationally expensive, here it's just a simple comparison, which is negligible. So I don't think this provides any value.

Reply:
Thanks @soltysh 
I understand what you mean, i close this PR now, thanks again!